### PR TITLE
Provide data in DELETE requests

### PR DESCRIPTION
**Pull request by Pimm**
This seems to work, and seems to be [in line with the PHP version](https://github.com/mollie/mollie-api-php/blob/f3ba2f3e776eb6b9ff0973e817282dd759c67d82/src/Endpoints/EndpointAbstract.php#L121).

Possibly related to #147.

### CODE REVIEW

File: src/resources/Resource.ts [hunk 1]
```diff
@@ -87,9 +87,9 @@ export default class Resource<R, T extends R> {
         }
         return this.injectPrototypes(response.data);
       },
-      delete: async <S extends T | true>(url: string): Promise<S> => {
+      delete: async <S extends T | true>(url: string, context: any): Promise<S> => {
```
Comment (vernondegoede) [inline src/resources/Resource.ts:90]: Does this need a default value?

Comment (Pimm) [reply src/resources/Resource.ts:90]: We could do `undefined` by default. I'm OK with that.

Comment (vernondegoede) [reply src/resources/Resource.ts:90]: It's also fine with leaving it like this.

Comment (Pimm) [reply src/resources/Resource.ts]: I thought about it a little more, and now I like the idea a little more. I just changed it.

**Comment by Pimm** [comment]
@vernondegoede Could you retrigger Travis? I think it failed to notify GitHub of the results.

**Pull request merged**